### DISCUSSION TITLE
Navigation in generated site is not working correctly, due to bug in index.ts

### DISCUSSION THREAD

**Discussion by selism** [mkdocs/mkdocs#3442] [Q&A] [answered]
When navigating through the site that is generated we notice that the navigation is not working well.
I managed to track down the issue due an error in the javascript files that handles the navigation in the site index: the `/assets/javascripts/src/templates/assets/javascripts/integrations/instant/index.ts` file.
In the generated site, that javascript file is bundled into `/assets/javascripts/bundle.aecac24b.min.js`.

Line 93 of that file reads as 
``` js
  canonical.href = canonical.href.replace("//localhost:", "//127.0.0.1")
```
As you can see, that line has a bug.
When the address `https://localhost:8000/` comes in, that is translated to `https://127.0.0.1800/`, which is an invalid url.
This bug has the strange effect that when clicking on some links, the address bar in the browser is updated to reflect the url of the new page, but the browser itself does not navigate to that page.

**Comment by pawamoy** [comment]
That's an issue for Material for MkDocs it seems.

**Reply by squidfunk** [reply, marked as answer]
@pawamoy is correct! We just fixed it in [`87d69a9`](https://github.com/squidfunk/mkdocs-material/commit/87d69a95b8284d60233d94563953ecf7b1dfc7ec).
